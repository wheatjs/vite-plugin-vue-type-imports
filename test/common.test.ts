import type { CodeGetter } from './_utils'
import { defineTransformTest } from './_utils'

const codeGetter: CodeGetter = ({ entry }) => `<script lang="ts" setup>
import { Props } from '${entry}'

defineProps<Props>()
</script>
`

defineTransformTest({
  category: 'Common',
  filePattern: ['./fixtures/common/**/!(_)*.ts'],
  fileName: __filename,
  codeGetter,
})
