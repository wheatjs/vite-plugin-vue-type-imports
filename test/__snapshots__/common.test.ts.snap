// Vitest Snapshot v1

exports[`Common > Interface extends interface > No reference > index.ts (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    baz: boolean;

    foo: string;
    bar: number;
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Interface extends interface > No reference > index.ts (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    baz: boolean;

    foo: string;
    bar: number;
}



defineProps<Props>()

</script>
"
`;

exports[`Common > Interface extends interface > No reference > index.ts (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    baz: boolean;

    foo: string;
    bar: number;
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Interface extends interface > No reference > index.ts (default) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    baz: boolean;

    foo: string;
    bar: number;
}



defineProps<Props>()

</script>
"
`;

exports[`Common > Interface without reference > Default > index.ts (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    foo: string;
    bar: number;
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Interface without reference > Default > index.ts (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    foo: string;
    bar: number;
}



defineProps<Props>()

</script>
"
`;

exports[`Common > Interface without reference > Default > index.ts (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    foo: string;
    bar: number;
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Interface without reference > Default > index.ts (default) 1`] = `
"<script lang=\\"ts\\" setup>
interface Props {
    foo: string;
    bar: number;
}



defineProps<Props>()

</script>
"
`;

exports[`Common > Redeclaration of types > Default > index.ts (clean all) 1`] = `
"<script lang=\\"ts\\" setup>
type Foo = [number, number]
interface Props {
    foo: Foo
    bar: Foo
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Redeclaration of types > Default > index.ts (clean interface) 1`] = `
"<script lang=\\"ts\\" setup>
type Foo = [number, number]
interface Props {
    foo: Foo
    bar: Foo
}



defineProps<Props>()

</script>
"
`;

exports[`Common > Redeclaration of types > Default > index.ts (clean newline) 1`] = `
"<script lang=\\"ts\\" setup>
type Foo = [number, number]
interface Props {
    foo: Foo
    bar: Foo
}
defineProps<Props>()
</script>
"
`;

exports[`Common > Redeclaration of types > Default > index.ts (default) 1`] = `
"<script lang=\\"ts\\" setup>
type Foo = [number, number]
interface Props {
    foo: Foo
    bar: Foo
}



defineProps<Props>()

</script>
"
`;
